### DISCUSSION TITLE
win10 compile weasis4.0.1 record and share

### DISCUSSION THREAD

**Discussion by runqinshiye** [nroduit/Weasis#294] [Show and tell]
[Win10专业版编译Weasis4.0.1环境配置220618.pdf](https://github.com/nroduit/Weasis/files/8932900/Win10.Weasis4.0.1.220618.pdf)

**Comment by nroduit** [comment]
Could you please tell me what is the result of: machine=$(echo "${ARC_OS}" | cut -d'-' -f1)

If there is a bug, it would be better to fix it than replacing by a specific value.

**Reply by runqinshiye** [reply]
thanks your great work.
i am a windows developer,so i understand  windows bat, don't know sh. so i direct modify the sh file.
if don't modify sh file $machine value , git bash excute error:
![J BYA)A25D%0WR(2N(LG4}C](https://user-images.githubusercontent.com/76540623/174431022-bd6b7740-ad6a-41d6-9a5c-78e6013de9ce.png)

**Reply by runqinshiye** [reply]
i try again,find modify one location to compile is ok,
modify this 👇
![A DI{(6 HPD(S4R%0 L}IJX](https://user-images.githubusercontent.com/76540623/174431331-a0c06d76-6446-4dce-bb7e-2a1e2178d70f.png)
git bash to complie pass 👇
![S)T UK4`(HMIM{8GE`QOIH](https://user-images.githubusercontent.com/76540623/174431342-f4c139a6-8e81-4ca5-b65d-dc4b44cbbff2.png)

**Reply by nroduit** [reply]
I guess the issue is related to Windows backslash path. Escaping `$` by another `\` could help, try to replace by `RES="${curPath}\resources\\$machine"`

**Reply by runqinshiye** [reply]
thanks your guild for this issue,i modify sh file and compile is ok, thans very much.
![MKIEZ04PSP3($3 93LF2R~B](https://user-images.githubusercontent.com/76540623/174434546-2bb4dc85-7a7e-4637-95f9-005680318639.png)

**Reply by nroduit** [reply]
I published a [fix](https://github.com/nroduit/Weasis/issues/296). Let me know if it is working correctly now.

**Reply by runqinshiye** [reply]
i [modify one location](https://github.com/nroduit/Weasis/discussions/294#discussioncomment-2976210) is ok , your fix can compile, but install after start faild.

**Reply by nroduit** [reply]
Try to update the git repository because there were some lines for another bug.

**Reply by runqinshiye** [reply]
i sync git master repository and compile,confirm issue is not exit,thank you for your greate work
![4E%ML% WIGWSGDK4XEFWVAD](https://user-images.githubusercontent.com/76540623/174475539-4075b51d-95cf-4daa-8fbc-f295a1886c2b.png)

i have a new issue：the  weasis protocol  invalid,i don't find reason for i made the installation package locally.

**Reply by dev45321** [reply]
i guess for the the weasis protocol invalid, u can follow this thread https://groups.google.com/g/dcm4che/c/pMCWBJjVUQM, and if possible modify the sh file in the right way and then create a push
thanks!!

**Reply by runqinshiye** [reply]
thanks for your guild,i modify and compile is ok,issue is not exit. i submit a pull request,please review.

**Reply by runqinshiye** [reply]
install the weasis,cmd and chrome use weasis protocol, the weasis start ok ,but can not open remote dcm image,

The error screenshot is as follows：
start weasis://%24dicom%3Aget+-w+%22https%3A%2F%2Fnroduit.github.io%2Fsamples%2FLumbar%2Fmf.xml%22
![S_2PAWSRHL8LQR15T }O%IM](https://user-images.githubusercontent.com/76540623/174479222-58c152ef-6fc4-4361-8549-89683d098bdd.png)
![0)3IXC4GV19Q_V(W3` @SJE](https://user-images.githubusercontent.com/76540623/174479224-a5f851f4-c050-4fb0-83a5-42fd025db5e3.png)

**Comment by nroduit** [comment]
Do you have the same issue when installing the official build?

It could be clock synchronization of your computer or cacerts not installed with jdk.

**Reply by runqinshiye** [reply]
thank you,I found the cause of the problem because I used a proxy and the certificate was incorrect.
